Tasks/S3Upload/TaskOperations.ts

```diff
@@ -79,6 +96,13 @@ export class TaskOperations {
         )
 
         const matchedFiles = this.findMatchingFiles(this.taskParameters)
+
+        if (matchedFiles.length === 0) {
```

Comment:
This seems like a breaking change to have the task fail if no files are found to be uploaded. 

I can imagine in a pipeline that some previous task might sometimes cause files to be upload but not always. The task would always run on the previous task's output folder and upload any files if there were any. Then continue with the pipeline. Now with this change the pipeline would fail when there were no files to upload from the previous task.

Reply:
Makes sense, I'll change this to a warning because I think it should at least be messaged that nothing was uploaded.